File: 프로그래머스/0/120890. 가까운 수/가까운 수.java
class Solution {
    public int solution(int[] array, int n) {
        int answer = Integer.MAX_VALUE;
        
        for(int i = 0; i < array.length; i++){
            if(Math.abs(array[i] - n) < Math.abs(answer - n)) {
                answer = array[i];
            }else if(Math.abs(array[i] - n) == Math.abs(answer - n)){
                answer = array[i] < answer ? array[i] : answer;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120890. 가까운 수/가까운 수.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// array_len은 배열 array의 길이입니다.
int solution(int array[], size_t array_len, int n) {
    int answer = 0;
    int diff = 101;
    
    for(int i = 0; i < array_len; i++){
        if(abs(array[i] - n) < diff){
            answer = array[i];
            diff = abs(array[i] - n);
        }else if(abs(array[i] - n) == diff){
            answer = array[i] < answer ? array[i] : answer;
        }
    }
    
    return answer;
}